src/CommunityToolkit.Maui.Core/Essentials/BadgeCounter/BadgeCounter.shared.cs

```diff
@@ -0,0 +1,20 @@
+﻿namespace CommunityToolkit.Maui.BadgeCounter;
+
+/// <inheritdoc cref="IBadgeCounter"/>
+public static class BadgeCounter
+{
+    static IBadgeCounter? defaultImplementation;
+
+    /// <inheritdoc cref="IBadgeCounter.SetBadgeCount" />
+	public static void SetBadgeCount(int count)
+    {
+        Default.SetBadgeCount(count);
+    }
+
+    /// <inheritdoc cref="IBadgeCounter" />
+	public static IBadgeCounter Default =>
+        defaultImplementation ??= new BadgeCounterImplementation();
+
+    internal static void SetDefault(IBadgeCounter implementation) =>
+        defaultImplementation = implementation;
```

Comment:
Let's use `Lazy<T>` to ensure thread safety.

Let's change the class name to `Badge`.

Let's use `SetCount()` now that the class name is `Badge`.

```suggestion
	static Lazy<IBadge> defaultImplementation = new(new BadgeImplementation());

	/// <inheritdoc cref="IBadge.SetCount" />
	public static void SetCount(int count)
	{
		Default.SetCount(count);
	}

	/// <inheritdoc cref="IBadge" />
	public static IBadge Default => defaultImplementation.Value;

	internal static void SetDefault(IBadge implementation) =>
		defaultImplementation = new(implementation);
```